### DISCUSSION TITLE
v28 30.06.2022

### DISCUSSION THREAD

**Discussion by dosinabox** [dosinabox/g2nr_unofficial_update#457] [Announcements]
- Запись о наводке на Снафа от Хуно добавляется в дневник только один раз.
- Исправлена невозможность начать квест 'Пакет с отмычками', если не спросить Фиска о нападении до встречи в таверне.
- Исправлены нелогичные диалоги и сброс местоположения Фиска, если его первая встреча с ГГ состоялась в таверне.
- Реализована реакция Сеньяна на убийство Эстебана и задействована его неиспользуемая реплика.
- Если ГГ не говорил с Сеньяном, то Эмилио не скажет, что слышал их разговор.
- Диалог с Эмилио об убийстве Сеньяна не добавит запись в неактивный квест 'Покушение на убийство'.
- Квест 'Ополос и библиотека' не начнется заново в диалоге с Ополосом, если ГГ уже высказал пожелание Пирокару.
- Дуриана можно спросить подробности его изгнания из монастыря, если ГГ не сделал этого до диалога о свитке сна.
- Диего правильно уйдет на точки ожидания у башни Ксардаса и шахты Сильвестро.
- При добавлении в дневник новых записей на экране отображается название обновленных квестов и рубрик.
- Квест 'Как попасть в монастырь' отменяется при вступлении в ополчение или к наемникам.
- Горакс начнет продавать соответствующие пояса только после вступления ГГ в послушники и маги.
- Горакс начнет продавать карту алтарей после вступления ГГ в маги.
- После диалога с Ультаром в торговом инвентаре Горакса не будет создаваться повторная карта алтарей.
- Исправлены ситуации, когда драки с наемниками считались нападением на крестьян со штрафами у Ли.
- Из прохода над пещерой испытания Огнем не будет выпадать краулер.
- Добавлена улучшенная модель кирки (взято из мода 'Хроники Миртаны: Архолос').
- Добавлена правильная анимация питья пива для ГГ (взято из мода 'Хроники Миртаны: Архолос').
- Исправлена сортировка свитков и рун паладинского исцеления в инвентаре.
- Исправлены зависания скелетов-магов в состоянии атаки (спасибо ElderGamer).
- В текстуре у входа в канализацию не будут застревать крысы.
- Исправлена непроницаемая трава около входа в дом Онара.
- В информации о сохранении и в дневнике отображается номер текущей главы (только с Union, спасибо Slavemaster).
- Изменена точка ожидания Ангара на Ирдорате, чтобы он не появлялся в трупе Феодарона.
- Добавлена возможность отключения прогрессивной стоимости обучения через помощника по сюжету.
- Добавлена возможность включения платного обучения через помощника по сюжету.
- ГГ больше не будет спотыкаться на помостах на выходах из города (спасибо Kvincius).
- Исправлены недоступные предметы в нескольких сундуках Хориниса и Яркендара (спасибо Kvincius).
- Модель чучела головы мракориса использует правильную текстуру.
- Валентино, Саландрил и судья будут реагировать на использования секретных переключателей в их домах.
- Исправлена возможность получения 20 очков силы с помощью амулета силы и кольца предводителей орков.
- Квест 'Секретная винокурня' не начнется в диалоге с Энимом в третьей главе, если ГГ уже открыл решетку ранее.
- Исправлен разрыв в модели тяжелой робы сектанта у Фортуно.
- Лестер теперь носит среднюю робу сектанта, как в Готике 1.
- День получения лекарства от одержимости у Пирокара не влияет на дневной лимит Ватраса на Ирдорате.
- Аллигатор Джек теперь правильно собирает мясо со всех болотных крыс.
- В рамках квеста 'Охота на бритвозубов' нельзя привести пиратов на пляжи около пиратского лагеря.
- В раздел дневника 'Карманная кража' добавляется информация об украденных предметах и золоте.
- Исправлена блокировка карманной кражи у Моргахарда.
- Характеристики Гарвига скорректированы с учетом аддона: теперь он сильнее остальных послушников.
- За убийство или избиение Гарвига ГГ получает соответствующее количество опыта вместо 10.
- Путь на Ирдорат занимает более суток внутриигрового времени: переход не происходит мгновенно.
- В торговый инвентарь Мигеля добавлено несколько трав.
- Стражники восстанавливают стандартные реакции и восприятия после пропуска ГГ.
- Добавлено меню подтверждения начала новой игры.
- Задействовано несколько неиспользуемых анимаций для стражников.
- Рабы и каторжники не реагируют на оружие и магию ГГ.
- Исправлено зависание экипированных рун при полном сбросе через помощника по персонажу.
- После диалога с Акилом или Кати Рэндольф не вернется на ферму, если находится в таверне на состязании.
- Исправлено соотношение сторон рисунка Бабо.

<hr /><em>This discussion was created from the release <a href='https://github.com/dosinabox/g2nr_unofficial_update/releases/tag/v28'>v28 30.06.2022</a>.</em>

**Comment by Kor-Angar** [comment]
> Лестер теперь носит среднюю робу сектанта, как в Готике 1.

зачем? :-)

**Reply by Kor-Angar** [reply]
стрёмный Лестер :-)
![image](https://user-images.githubusercontent.com/49514764/177461551-fbdbb22b-9cda-40e4-be2f-76611561c2f4.png)

**Comment by randoMsr** [comment]
1) Если взять кв "Услуга Кантару" во второй или более поздней главе, пропадает возможность сообщить Кантару о смерти Сары, так же было в v27
2) Альрик становится сильней в старших главах, но только после того как скажет ГГ что за это время он стал лучше,
но из-за исправления:
- Начиная с третьей главы Альрик не будет говорить о повторном бое с ГГ, если ни разу не сражался с ним.
в любой главе, в первом бою Альрик будет 10го уровня (не знаю, нуждается ли в исправлении)

**Reply by Kor-Angar** [reply]
про Альрика не понятно. там его уровень растёт в Готики 2 классик, в том числе уровень владения оружием. в СкиптПатче @dosinabox  учителям ставила навык владения оружием равный уровню тренировку.

**Reply by dosinabox** [reply]
> в СкиптПатче @dosinabox учителям ставила навык владения оружием равный уровню тренировку.

Но не Альрику.

> там его уровень растёт в Готики 2 классик, в том числе уровень владения оружием.

Точно так же растет и в Г2НВ и в обновлении, два раза: 40 уровень в 3/4 главе и 60 в 5. Оба раза происходит и повышение одноручного навыка на 20%.

> в любой главе, в первом бою Альрик будет 10го уровня (не знаю, нуждается ли в исправлении)

Наверное, не нуждается. Критерий повышения это не только глава, но количество боев (1-6 и 1-9).
Видимо, имеется в виду проблема, когда ГГ впервые сражается уже в поздних главах и Альрик оказывается слишком слабым? Тогда надо как-то учитывать главу первой драки и динамически рассчитывать момент повышения навыков и уровня Альрика. И повышать их вне диалога.

**Reply by randoMsr** [reply]
Альрик после первого боя обращается к ГГ и моментально становится сильней (если в 5ой главе - моментальное повышение с 10 на 60 лвл)

**Reply by dosinabox** [reply]
Тогда надо повышать уровень и статы по главам (1-10, 2-20, 3-30, 4-40, 5-50) без использования диалогов. А обращение делать в следующей главе после первого боя.

**Reply by Kor-Angar** [reply]
сейчас повышение как в в оригинале в 3 и 5 главе? 

```D
FUNC INT DIA_Alrik_NewFights3_Condition()
{
	if (Kapitel >= 3)
	&& (Kapitel <= 4)
	&& (Alrik_ArenaKampfVerloren <= 6)
	{
		return TRUE;
	};
};

FUNC VOID DIA_Alrik_NewFights3_Info()
{
	B_AddFightSkill (self, NPC_TALENT_1H, 20);
	B_SetAttributesToChapter (self, 4);
	B_Alrik_Again();
};
```

```D
FUNC INT DIA_Alrik_NewFights5_Condition()
{
	if (Kapitel >= 5)
	&& (Alrik_ArenaKampfVerloren <= 9)
	{
		return TRUE;
	};
};

FUNC VOID DIA_Alrik_NewFights5_Info()
{
	B_AddFightSkill (self, NPC_TALENT_1H, 20);
	B_SetAttributesToChapter (self, 6);
	B_Alrik_Again();
};
```

**Reply by Kor-Angar** [reply]
> Альрик после первого боя обращается к ГГ и моментально становится сильней (если в 5ой главе - моментальное повышение с 10 на 60 лвл)

лучше не делать очередной новодел, а доделать логику пираней  взяв условия отсюда

```D
	// ----- EXIT: ЬBER DREI MAL Alrik_ArenaKampfVerloren ------
	else if (Kapitel <= 2)
	&& (Alrik_ArenaKampfVerloren > 3)
	{
		AI_Output (self, other, "DIA_Alrik_Add_09_00"); //Ich denke fьr's erste reicht es...
		AI_Output (self, other, "DIA_Alrik_Add_09_01"); //Gцnn mir mal ne kurze Pause.
		B_Alrik_ComeBackLater();
	}
	
	else if (Kapitel >= 3)
	&& (Kapitel <= 4)
	&& (Alrik_ArenaKampfVerloren > 6)
	{
		B_Alrik_Enough();
		B_Alrik_ComeBackLater();
	}
	
	else if (Kapitel >= 5)
	&& (Alrik_ArenaKampfVerloren > 9)
	{
		B_Alrik_Enough();
		AI_Output (self, other, "DIA_Alrik_Add_09_04"); //AuЯerdem habe ich mittlerweile genug Geld zusammen.
		AI_Output (self, other, "DIA_Alrik_Add_09_05"); //Ich hцr auf und such' mir hier irgendwo in der Stadt 'ne Bleibe...
		AI_Output (self, other, "DIA_Alrik_Add_09_06"); //Mal sehen, vielleicht mach in 'nen Waffenladen auf...
	}
```

и добавив проверку на минимальное кол-во боёв сюда

```D
FUNC INT DIA_Alrik_NewFights5_Condition()
{
	if (Kapitel >= 5)
	&& (Alrik_ArenaKampfVerloren <= 9)
	&& (Alrik_ArenaKampfVerloren > 6)
	{
		return TRUE;
	};
};
```


```D
FUNC INT DIA_Alrik_NewFights3_Condition()
{
	if (Kapitel >= 3)
	&& (Kapitel <= 4)
	&& (Alrik_ArenaKampfVerloren <= 6)
	&& (Alrik_ArenaKampfVerloren > 3)
	{
		return TRUE;
	};
};
```

**Reply by Kor-Angar** [reply]
то есть на выходе на сколько я понимаю должна быть такая логика
![image](https://user-images.githubusercontent.com/49514764/177030689-9b27141e-37fe-4304-9731-5ae8f0b31f7c.png)

**Reply by dosinabox** [reply]
> сейчас повышение как в в оригинале в 3 и 5 главе?

Да.

> добавив проверку на минимальное кол-во боёв сюда

Alrik_ArenaKampfVerloren надо > 2 и > 5.
И всё равно - если третий бой произойдет в 3 или 4 главе (или шестой в 5), то Альрик тут же объявит, что стал сильнее.
```cpp
AI_Output(self,other,"DIA_Alrik_Add_09_03");	//Как дела? Ты хочешь сразиться со мной еще раз? Я думаю, за это время я стал лучше...
```

**Reply by Kor-Angar** [reply]
можно в `DIA_Alrik_NewFights3_Condition` добавить проверку на `(Npc_IsInState(self,ZS_Talk))`, что бы это реплика шла при обращение героя в самом начале диалога.

***

в саму реплику `B_Alrik_Again `можно поставить доп проверку что бы выводить её или не выводить в зависимости от того была ли тематическая реплика Альрика  в предыдущей главе в  `DIA_Alrik_WannaFight_Info `



что то типа
```D

	else if((Kapitel <= 2) && (Alrik_ArenaKampfVerloren > 3))
	{
   Alrik_Again_2022 = true;

		AI_Output(self,other,"DIA_Alrik_Add_09_00");	//Я думаю, пока хватит...
		if(Wld_IsTime(11,0,19,0))
		{
			AI_Output(self,other,"DIA_Alrik_Add_09_02");	//Заходи позже. А пока я потренируюсь...
		}
		else
		{
			AI_Output(self,other,"DIA_Alrik_Add_09_01");	//Дай мне немного передохнуть.
		};
	}
	else if((Kapitel >= 3) && (Kapitel <= 4) && (Alrik_ArenaKampfVerloren > 6))
	{
   Alrik_Again_2022 = true;
		B_Alrik_Enough();
		AI_Output(self,other,"DIA_Lehmar_NOCHMALGELD_09_03");	//Заходи через несколько дней!
	}
```

```D
func void B_Alrik_Again()
{
    if Alrik_Again_2022 == true
    {
	AI_Output(self,other,"DIA_Alrik_Add_09_03");	//Как дела? Ты хочешь сразиться со мной еще раз? Я думаю, за это время я стал лучше...
	Alrik_Again_2022 = false;
    };
};
```

**Reply by dosinabox** [reply]
Чтобы не было моментального повышения сразу после боя, можно запоминать главу и день последнего сражения. Если глава сражения равна текущей, то тогда проигрывать диалоги с повышениями только через 2 дня.

**Reply by Kor-Angar** [reply]
тогда у тебя он может и не повысится, так как за это время может "проскочить" условие по кол-ву боёв ;-)

**Reply by Kor-Angar** [reply]
если ты хочешь прям идеально тогда надо переписывать этот фрагмент убираю оттуда в том числе проверку на 1 и 2 главу и добавляя соответствующие маркеры. маркер я показал как пример его наверное как то и более корректно можно сделать.
```D
else if((Kapitel <= 2) // убирая эту проверку
&& (Alrik_ArenaKampfVerloren > 3))
	{
   Alrik_Again_2022 = true; // добавляя маркер

		AI_Output(self,other,"DIA_Alrik_Add_09_00");	//Я думаю, пока хватит...
		if(Wld_IsTime(11,0,19,0))
		{
			AI_Output(self,other,"DIA_Alrik_Add_09_02");	//Заходи позже. А пока я потренируюсь...
		}
		else
		{
			AI_Output(self,other,"DIA_Alrik_Add_09_01");	//Дай мне немного передохнуть.
		};
	}
	else if((Kapitel >= 3) && (Kapitel <= 4) && (Alrik_ArenaKampfVerloren > 6))
	{
   Alrik_Again_2022 = true;
		B_Alrik_Enough();
		AI_Output(self,other,"DIA_Lehmar_NOCHMALGELD_09_03");	//Заходи через несколько дней!
	}
```
таким образом связывая повышение напрямую с отказом сражаться.
то есть без отказа сражаться не должны быть и соответствующей реплики.  иначе может появится баг в результате которого герой сможет побить слабого альрика много раз.

**Reply by dosinabox** [reply]
Тогда если глава и день сражения равны текущим, то использовать альтернативную подрезанную реплику:
```cpp
AI_Output(self,other,"DIA_Alrik_Add_09_03_add");	//Ты хочешь сразиться со мной еще раз? Я думаю, я стал лучше...
```

**Reply by Kor-Angar** [reply]
или тупо оставить как сейчас :-) я практически ничего сказать не могу так как после его твоего фикса вообще перестал драться с Альриком а просто забирать его меч ;-)

**Reply by dosinabox** [reply]
а чем фикс стал мешать твоим дракам с Альриком?

**Reply by Kor-Angar** [reply]
> Тогда если глава и день сражения равны текущим, то использовать альтернативную подрезанную реплику:

такое точно надо в игре проверять :-) он там еще эликсирами подряд закидывается и надо проверять что бы при повышение его параметров после боя и повторной битве всё билось и по жизням и по навыкам, так как у него может непереключится боевая стойка. то есть что бы это не превратилось в простое избиение.

***

могу предложить так же за него повторно выдавать опыт. так как он сначала одного уровня, а потом другого.

**Reply by Kor-Angar** [reply]
> а чем фикс стал мешать твоим дракам с Альриком?

он мне мешает. я же редко предлагаю что лично мне помогает играть.
 в данном случае эта правка мне помогает делаю игру для меня легче, так как я могу купить меч не сражаясь с Альриком. то есть низкоуровневым героем. что я и делаю. беру его то ли на втором то ли на третьем уровне. то есть в первый день в городе. раньше я его брал позже после того как мог побить альрика. а я же не мастер дуэлей играю на клаве и мне это было тяжело сделать на втором-третьем уровне :-)

**Reply by Kor-Angar** [reply]
> я же редко предлагаю что лично мне помогает играть.

то есть я писал о том что эта правку упростит в данном случае игровой процесс :-)

**Reply by dosinabox** [reply]
Какая правка?

**Reply by Kor-Angar** [reply]
я не помню деталей, очень жарко сегодня, а дождь никак не пойдёт :-)  у нас топик про Альрика был. суть в том что меч покупается/получается без драки с Альриком.

**Reply by Kor-Angar** [reply]
> я не помню

деталей

**Reply by dosinabox** [reply]
Что-то я этого тоже не помню. Здесь тоже жарко сегодня, сложно усваивать информацию.

> могу предложить так же за него повторно выдавать опыт. так как он сначала одного уровня, а потом другого.

А вот это хорошо, сделаю.

**Reply by Kor-Angar** [reply]
> А вот это хорошо, сделаю.

может тогда доп опыт. за вычетом предыдущего? :-)

***

- 100 (10 lvl) 
- 400-100 (40 lvl) 
- 600 - 100 -300 (60 lvl)

т.е. +100 + 300 +200

***

хотя с точки зрения ЛОРа наверное правильнее +100+200+300 ...

**Comment by randoMsr** [comment]
Если спросить Гаана как охота до того как он подойдет к точке с трупом фыркающего зверя, получим  опыт, но обещанную награду Гаан не даст. 
Демонстрация: https://youtu.be/vJRyf_Swi14
Если убиваем быстро, обязательно нужно ждать, пока Гаан дойдет до точки, иначе денег не получим

**Reply by dosinabox** [reply]
> Если спросить Гаана как охота до того как он подойдет к точке с трупом фыркающего зверя, получим опыт, но обещанную награду Гаан не даст.

исправлено, спасибо

**Reply by randoMsr** [reply]
Это Вам спасибо!

**Reply by Kor-Angar** [reply]
> получим опыт, но обещанную награду Гаан не даст.

оно так и задумано было. Что бы получить золото нужно договорится об этом с Гааном заранее. обычная механика. договорился о награде и получил награду, опыт и хорошее отношение. не договорился, получил опыт и хорошее отношение.

**Reply by dosinabox** [reply]
> оно так и задумано было. Что бы получить золото нужно договорится об этом с Гааном заранее.

В диалоге "Как охота?" нет проверки на этот договор (MIS_Gaan_Deal или Gaan_Deal). И в видео ГГ таки спрашивает о награде:
![image](https://user-images.githubusercontent.com/39790055/179405410-23c91c37-e40f-47b3-abe9-1b6815a4a395.png)

**Reply by Kor-Angar** [reply]
извините. видимо не правильно понял. решил что речь про эту проверку.

```D
	if (MIS_Gaan_Deal == LOG_RUNNING)
		{
			AI_Output			(self, other, "DIA_Gaan_MONSTERTOT_03_02");	//Hier hast du das versprochene Geld.

			CreateInvItems (self, ItMi_Gold, 30);									
			B_GiveInvItems (self, other, ItMi_Gold, 30);					
		};
```

**Reply by Kor-Angar** [reply]
проверки на диалог в MDK

```D
func int DIA_Gaan_MONSTERTOT_Condition ()
{
	if 	((Npc_IsDead(Gaans_Snapper))== TRUE) 
	&& 	(RangerMeetingRunning != LOG_RUNNING)
			{
					return TRUE;
			};
};
```
